apps/web/vibes/soul/components/counter/index.tsx

```diff
@@ -1,58 +1,51 @@
-'use client'
+import { DecrementButton } from './decrement-button'
+import { IncrementButton } from './increment-button'
 
-import { useState } from 'react'
+// This is the Counter Submission Component
 
-import { Minus, Plus } from 'lucide-react'
+// TODO: Since we need a Counter with an input and no submit buttons on the PDP page, should I create another reusable component? or should that be a one off instance and this remains the library component?
 
 interface Props {
+  id: string
   current?: number
   max?: number
   type?: 'button' | 'submit'
+  updateLineItemQuantityAction({ id, quantity }: { id: string; quantity: number }): Promise<void> //formData.get('id'), formData.get('quantity')
 }
 
-export const Counter = function Counter({ current = 0, max = 20, type = 'button' }: Props) {
-  const [count, setCount] = useState(current)
-
+export const Counter = function Counter({
+  id,
+  current = 0,
+  max = 20,
+  type = 'button',
+  updateLineItemQuantityAction,
+}: Props) {
   const decrement = () => {
-    if (count > 0) {
-      setCount(prev => prev - 1)
+    if (current > 1) {
+      return current - 1
+    } else {
+      return current
```

Comment:
Allow settings 0 or negatives. It's up to the one handling the actions to remove item if needed.

Reply:
@jorgemoya should I add an optional `min` prop?